src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,169 @@
+                int pos = Integer.parseInt(input.substring(7).trim());
+                if (pos > tasks.getSize() || pos == 0) {
+                    throw new DukeException("\tThis number is out of bounds! ");
+                }
+                Task element = tasks.getTask(pos - 1);
+                tasks.deleteTask(pos - 1);
+                ui.printDeleteTasks(pos, tasks, element);
+                storage.writeTasksToFile(tasks);
+            } else if (input.startsWith("list")) {
+                ui.printListMessage(tasks);
+                storage.writeTasksToFile(tasks);
+            } else if (input.startsWith("find")) {
+                String keyword = input.substring(5).trim();
+                TaskList foundTasks = new TaskList();
+                for (int i = 0; i < tasks.getSize(); i++) {
+                    Task task = tasks.getTask(i);
+                    if (task.getDescription().contains(keyword)) {
+                        foundTasks.addTask(task);
+                    }
+                }
+
+                if (foundTasks.getSize() > 0) {
+                    ui.printFindTask(foundTasks);
+                } else {
+                    System.out.println("\tNo tasks found containing the keyword: " + keyword);
+                }
+            } else {
+                throw new DukeException("\tHey bud! Sorry I don't quite know what you mean :-(");
+            }
+        } catch (DukeException | DateTimeException | IOException e) {
+            handleException(e);
+        }
+    }
+
+
+    /**
+     * Deals with user commands to add a task to the tasklist.
+     * If the command is not one that adds a task, it will call the userCommand
+     * method.
+     *
+     * @param input what the user is typing in.
+     * @param storage the storage that is being used.
+     * @param tasks the TaskList that is being used to store the tasks.
+     */
+    public static void addToList(String input, Storage storage, TaskList tasks) {
+        try {
+            if (input.startsWith("todo")) {
+                if (input.trim().length() <= 4) {
+                    throw new DukeException("\t Sorry! The description of a todo cannot be empty :(");
+                }
+                Task task = new ToDo(input.substring(5));
+                tasks.addTask(task);
+                ui.printAddTaskToList(tasks, task);
+                storage.writeTasksToFile(tasks);
+            } else if (input.startsWith("deadline")) {
+                int index = input.lastIndexOf("/by");
+                if (input.trim().length() <= 8 || (input.substring(9, index).isEmpty())) {
+                    throw new DukeException("\t Sorry! The description of a deadline cannot be empty :(");
+                }
+
+                if (!input.contains("/by"))  {
+                    throw new DukeException("\t Hey bud! Please include when the deadline is! " +
+                            "\n\t For example you can type: deadline read /by 2023-09-01 1700");
```

Comment:
Would it be better to break a new line before operator '+'? 

coding standard link for your reference :) 
- https://se-education.org/guides/conventions/java/intermediate.html , 
- session: Layout > "Place line break to improve readability"